incubator/hnc/docs/user-guide/quickstart.md

```diff
@@ -0,0 +1,651 @@
+
+Imagine you have an org called _acme-org_. We'll create a _root_ namespace to
+represent it:
+
+```bash
+kubectl create namespace acme-org
+```
+
+The root namespace is just a normal namespace that we'll use as the root of our
+hierarchy. A K8s cluster may have many roots; the default case is that every
+namespace is the root of its own otherwise empty
+[subtree](concepts.md#basic-trees), with no parent or children.
+
+We can also create a namespace for a team within that org, and another namespace
+for a service owned by that team. The team might want different services in
+different namespaces, since [namespaces are a much better security
+boundary](concepts.md#why-ns) than, say, service accounts.
+
+```bash
+kubectl create namespace team-a
+kubectl create namespace service-1
+```
+
+By default, there's no relationship between these namespaces. For example, let's
+say that we want to make someone Site Reliability Engineer (SRE) for `team-a`,
+so we'll create an RBAC Role and a RoleBinding to that role.
+
+  Note: Typically, we'd create a rolebinding a user account, such as maya@example.com.
+  However, every distribution of K8s has different authentication systems, and
+  some (like [Kind](https://kind.sigs.k8s.io/)) don't have any at all.
+  Therefore, this quickstart uses K8s service accounts instead of user accounts,
+  but feel free to use any account that works on your cluster as you follow
+  along.
+
+```bash
+kubectl -n team-a create role team-a-sre --verb=update --resource=deployments
+kubectl -n team-a create rolebinding team-a-sres --role team-a-sre --serviceaccount=team-a:default
+```
+
+Similarly, we might want to have a super-SRE group across the whole org:
+
+```bash
+kubectl -n acme-org create role org-sre --verb=update --resource=deployments
+kubectl -n acme-org create rolebinding org-sres --role org-sre --serviceaccount=acme-org:default
+```
+
+Obviously, none of this affects `service-1`, since that's a completely
+independent namespace, and RBAC only applies at the namespace level:
+
+```bash
+kubectl -n service-1 get rolebindings
+```
+
+So this is where the HNC comes in. Let's make `acme-org` the parent of `team-a`,
+and `team-a` the parent of `service-1`.
+
+```bash
+# Make acme-org the parent of team-a
+kubectl hns set team-a --parent acme-org
+
+# This won't work, will be rejected since it would cause a cycle. Try it!
+kubectl hns set acme-org --parent team-a
+
+# Make team-a the parent of service-1
+kubectl hns set service-1 --parent team-a
+
+# Display the hierarchy
+kubectl hns tree acme-org
+```
+
+The output will be something like:
+
+```
+acme-org
+└── team-a
+     └── service-1
+```
+
+Now, if we check `service-1` again, we'll see that the rolebindings from the
+ancestor namespaces have been propagated to the child namespace.
+
+```bash
+kubectl -n service-1 describe roles
+# Output: you should see two roles
+
+kubectl -n service-1 get rolebindings
+# Output: you should see two rolebindings
+```
```

Comment:
Maybe specify "propagated from acme-org and team-a"

Reply:
Done